### PR TITLE
Add git.get_tag() and svn.get_tag()

### LINKED ISSUE #3987: Support for git describe --tags in tools.Git
In our CI release flow, we'd like to number Conan packages based on a git tag applied in our repo.

Right now if I apply a tag ```0.0.8```, then fetch that and and use ```git branch``` I see:

```
$ git branch
* (HEAD detached at 0.0.8)
```

whereas if I use ```git describe --tags``` I see:

```
0.0.8
```

I'd like to be able to use ```0.0.8``` as the Conan package version.

If I try using the example from https://docs.conan.io/en/latest/howtos/capture_version.html, namely:

```
return "%s_%s" % (git.get_branch(), git.get_revision())
```

on that checked out tag I see:

```
E:\DevelopBp\mmaguir1\src\nx\replicant>conan create . PORT/stable
version[HEAD (no branch)_0dfb63c9c341536e528ef63f2e66746cf83c8539]
ERROR: Value provided for package version, 'HEAD (no branch)_0dfb63c9c341536e528ef63f2e66746cf83c8539' (type Version), is too long. Valid names must contain at most 51 characters.
```

Alternatively, if there's some magic combination of the existing [Git tools](https://docs.conan.io/en/latest/reference/tools.html#tools-git) that will let me retrieve that tag name, that would be much appreciated!

Related:
https://github.com/conan-io/conan/issues/548 "generating version from git"
https://github.com/conan-io/conan/issues/1764 "Add Git helpers"

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [v1.8.4 Windows Server 2012 git 2.12.1] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Feature: Added ``get_tag()`` methods to ``tools.Git()`` and ``tools.SVN()`` helpers.
Changelog: Fix: ``get_branch()`` method of ``tools.SVN()`` helper now returns only the branch name, not the tag when present.
Docs: https://github.com/conan-io/docs/pull/1020

@TAGS: svn, slow

- [x] Refer to the issue that supports this Pull Request: closes #3987 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: conans/test/unittests/util/tools_test.py [hunk 1]
```diff
@@ -1840,7 +1875,30 @@ def test_branch(self):
...
+        svn.checkout(url='/'.join([project_url, 'prj1', 'branches', 'my_feature']))
+        self.assertIsNone(svn.get_tag())
+
+        svn = SVN(folder=self.gimme_tmp())
+        svn.checkout(url='/'.join([project_url, 'prj1', 'branches', 'issue3434']))
+        self.assertIsNone(svn.get_tag())
+
+        svn = SVN(folder=self.gimme_tmp())
+        svn.checkout(url='/'.join([project_url, 'prj1', 'tags', 'v12.3.4']))
+        self.assertEqual("tags/v12.3.4", svn.get_tag())
```
[the remark this answers is not in the document]
Comment (danimtb) [reply conans/test/unittests/util/tools_test.py]: oh, ok. Missed that detail in the issue comment feed. Thanks!

[the remark this answers is not in the document]
Comment (danimtb) [reply conans/test/unittests/util/tools_test.py]: Done

File: conans/test/unittests/util/tools_test.py [hunk 2]
```diff
@@ -1646,6 +1646,41 @@ def build(self):
...
+        self.assertIsNone(tag)
+
+    def test_in_tag(self):
+        """
+        Current checkout is on a tag
+        """
+        git = Git(folder=self.folder)
+        git.run("tag 0.0.0")
+        tag = git.get_tag()
+        self.assertIn("0.0.0", tag)
```
Comment (jgsogo) [inline conans/test/unittests/util/tools_test.py]: equals?

File: conans/test/unittests/util/tools_test.py [hunk 3]
```diff
@@ -1646,6 +1646,41 @@ def build(self):
...
+        """
+        Tag is defined but current commit is ahead of it
+        """
+        git = Git(folder=self.folder)
+        git.run("tag 0.0.0")
+        save(os.path.join(self.folder, "file.txt"), "")
+        git.run("add .")
+        git.run("commit -m \"new file\"")
+        tag = git.get_tag()
+        self.assertIsNone(tag, "0.0.0-1-\w{8}")
```
Comment (jgsogo) [inline conans/test/unittests/util/tools_test.py]: uhmmm... the `"0.0.0-1-\w{8}"` now is the message in case of failure :/

**Comment by jgsogo** [comment]
@danimtb, what about the test that raises from `get_branch` and/or `get_tag`?

**Comment by jgsogo** [comment]
This PR is breaking previous behavior. Although I'm agree that that current one is the good one, I want to be sure that you are aware of it (ping @conan-io/barbarians):

https://github.com/conan-io/conan/pull/4306/files#diff-4d836661d105934362de1c311c59ce8cR1882

**Comment by danimtb** [comment]
Yes, it was not consistent with `git.get_branch()` behavior, which is only returning the version name

**Comment by lasote** [comment]
Thanks for the observation @jgsogo. I also agree it is better now. Fortunately, scm is still experimental because only the usage brings the knowledge.

**Comment by jgsogo** [comment]
SCM is experimental, `tools.SVN` isn't (https://docs.conan.io/en/latest/reference/tools.html#tools-svn).... but, I vote for changing it ✅

**Comment by lasote** [comment]
It is experimental, change the docs.

**Pull request merged**